cpu/native/periph/timer.c

```diff
@@ -27,6 +27,8 @@
 #include <mach/mach_init.h>
 #include <mach/mach_port.h>
 #include <mach/mach_host.h>
+/* Both OS X and RIOT typedef thread_t. timer.c does not use either thread_t. */
+#define thread_t riot_thread_t
```

Comment:
`#define thread_t riot_thread_t` means “replace every occurrence of `thread_t` after this line to `riot_thread_t`”, so it replaces `typedef struct _thread thread_t;` in `RIOT/core/include/sched.h` to `typedef struct _thread riot_thread_t;` because `sched.h` is included after the line.

The essence of the problem is:
```C
/* mach_types.h */
typedef mach_port_t             thread_t;

/* sched.h */
typedef struct _thread thread_t;
```

This PR works around by:
```C
/* mach_types.h */
typedef mach_port_t             thread_t;

#define thread_t riot_thread_t

/* sched.h */
typedef struct _thread thread_t;
```

that results in:

```C
/* mach_types.h */
typedef mach_port_t             thread_t;

/* sched.h */
typedef struct _thread riot_thread_t;
```